lib/swiftner/API/upload.rb

```diff
@@ -41,6 +41,29 @@ def self.create(attributes)
         build(response.parsed_response) if response.success?
       end
 
+      # Creates an upload from local file.
+      # @param [Hash] attributes
+      # @option attributes [String] :upload_language (required)
+      # @option attributes [String] :upload_title (optional)
+      # @option attributes [String] :upload_description (optional)
+      # @option attributes [String] :upload_is_playlist (optional)
+      # @param [String] file_path
+      def self.create_from_file(attributes, file_path)
+        validate_required(attributes, %i[upload_language])
+        validate_language(attributes, :upload_language)
+        validate_file(file_path)
+
+        attributes[:upload_file_extension] = File.extname(file_path).delete_prefix(".")
+        signed_url = client.post(
+          "/upload/generate_signed_url",
+          body: attributes.to_json,
+          headers: { "Content-Type" => "application/json" }
+        )["url"]
+
+        client.put(signed_url, body: File.read(file_path), headers: { "Content-Type" => "application/octet-stream" })
+        # TODO: get created upload id and find it and return it
```

Comment:
I'll wait with a review until it's fully ready.

Looking good so far though!

Reply:
The API generates a signed URL for uploading but doesn't return an upload ID. The upload is created once the file is uploaded to the signed URL, so the gem doesn't have access to the upload ID. There were discussions about having the API create an upload with a status of 'waiting for file' when creating signed_url.